packages/jest-cli/src/runTest.js

```diff
@@ -13,74 +13,91 @@ import type {Path, Config} from 'types/Config';
 import type {TestResult} from 'types/TestResult';
 import type {Resolver} from 'types/Resolve';
 
+const fs = require('fs');
+const {getTestEnvironment} = require('jest-config');
 const BufferedConsole = require('./lib/BufferedConsole');
+const promisify = require('./lib/promisify');
 const {
   Console,
   NullConsole,
   setGlobal,
 } = require('jest-util');
 const getConsoleOutput = require('./reporters/getConsoleOutput');
 
+const testEnvironmentPragmaRegex = /@testEnvironment\s+(.*)/;
+
 function runTest(path: Path, config: Config, resolver: Resolver) {
-  /* $FlowFixMe */
-  const TestEnvironment = require(config.testEnvironment);
-  /* $FlowFixMe */
-  const TestRunner = require(config.testRunner);
-  /* $FlowFixMe */
-  const ModuleLoader = require(config.moduleLoader || 'jest-runtime');
+  return promisify(fs.readFile)(path, 'utf8').then(data => {
```

Comment:
Because there is no other work that can be done in Jest at the same time, so there is no benefit in using async code for reading the file :)